src/frontend/components/header/header.ts

```diff
@@ -40,12 +40,22 @@ export class Header {
     }
   }
 
+  resetTheme() {
+    this.settingOpened = false;
+    document.removeEventListener('click', this.resetTheme.bind(this), true);
+  }
+
+  openSettings() {
+    document.addEventListener('click', this.resetTheme.bind(this), true);
```

Comment:
@igor-ka I want the document click within the context of this component not as a whole.